File: src/models/episode.ts
// constructor takes required fields
// chain builder for non required

import { EnclosureObject } from "rss";

// check what metadata to include
// https://validator.w3.org/feed/

export class Episode {
    title: string;
    description: string;
    url: string;
    pubdate: Date;
    image: string;
    enclosure: EnclosureObject;

    // duration etc
    

    constructor(title: string, description: string, url: string, pubdate: Date, image: string, enclosure: string) {
        this.title = title;
        this.description = description;
        this.url = url;
        this.pubdate = pubdate;
        this.image = image;
        this.enclosure = {url: enclosure, type: "audio/mpeg"};
      }
  }

File: src/main/java/com/ctkcoding/rssgen/model/Episode.java
package com.ctkcoding.rssgen.model;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder(toBuilder = true)
public class Episode {
  String title;
  String description;
  String url;
  LocalDateTime pubDate;
  String image;
  String enclosureUrl; // "{url: enclosure, type: \"audio/mpeg\"}"
  String enclosureType; // type: \"audio/mpeg\"
  Long enclosureLength;
  int duration;
}
